SeQuant/core/parse/parse.cpp

```diff
@@ -0,0 +1,219 @@
+//
+// Created by Robert Adam on 2023-09-20
+//
+
+#include "SeQuant/core/parse_expr.hpp"
+#include "ast.hpp"
```

Comment:
nonconforming #include, generic names like `ast.hpp` are likely to produce clashes ... please convert to `SeQuant/core/parse/ast.hpp`, here and elsewhere

Reply:
Already done in https://github.com/ValeevGroup/SeQuant/pull/142

Also, strictly speaking this is guaranteed to never include the wrong file as includes in quotes are always first searched in the current directory. It's only when it is not found there that it looks elsewhere in the include directories for it, at which point issues with name clashes can arise.

Reply:
> Also, strictly speaking this is guaranteed to never include the wrong file as includes in quotes are always first searched in the current directory. It's only when it is not found there that it looks elsewhere in the include directories for it, at which point issues with name clashes can arise.

The standard does not guarantee this: https://en.cppreference.com/w/cpp/preprocessor/include ...  but I might be overly paranoid

Reply:
Hm. I think I've read it somewhere, but I never actually checked the standard myself. Anyway, better safe than sorry :)